config/chain_config.js

```diff
@@ -0,0 +1,55 @@
+class ChainConfig {
```

Comment:
A proposal to evaluate: to make it readable and also to separate concepts, we can have to separate config representations for origin and auxiliary. Something like this:

config.origin.ipcProvider
config.origin.chainID
config.origin.contractAddresses.gateway

config.auxiliary.contractAddresses.coGateway